lib/features/modeling/behavior/CreateChoreoTaskBehavior.js

```diff
@@ -29,144 +29,172 @@ export default function CreateChoreoTaskBehavior(injector, bpmnFactory, canvas,
-
-      // properly link elements regarding choreography tasks
-      if (is(shape, 'bpmn:ChoreographyTask')) {
-        linkMessageFlowSemantics(injector, shape.businessObject, context.createdMessageFlow);
-        canvas.addShape(context.createdMessageShape, context.createdShapeA);
+      //Same as above: new vs. copied
+      if (!shape.oldBusinessObject) {
+        // add the bands to the BPMNDiagram semantic object
+        let choreo = canvas.getRootElement().businessObject;
+        let bpmnDiagram = choreo.di;
+        bpmnDiagram.get('planeElement').push(context.createdShapeA.diBand);
+        bpmnDiagram.get('planeElement').push(context.createdShapeB.diBand);
+        context.createdShapeA.diBand.$parent = bpmnDiagram;
+        context.createdShapeB.diBand.$parent = bpmnDiagram;
+
+        // add bands to canvas
+        canvas.addShape(context.createdShapeA, shape);
+        canvas.addShape(context.createdShapeB, shape);
+
+        // properly link elements regarding choreography tasks
+        if (is(shape, 'bpmn:ChoreographyTask')) {
+          linkMessageFlowSemantics(injector, shape.businessObject, context.createdMessageFlow);
+          canvas.addShape(context.createdMessageShape, context.createdShapeA);
+        }
+
       }
     }
   });
 
+
   this.reverted('shape.create', function(event) {
     let context = event.context;
     let shape = context.shape;
 
     if (is(shape, 'bpmn:ChoreographyActivity') && shape.type !== 'label') {
-      // remove the bands from the BPMNDiagram semantic object
-      let choreo = canvas.getRootElement().businessObject;
-      let bpmnDiagram = choreo.di;
-      bpmnDiagram.get('planeElement').pop();
-      bpmnDiagram.get('planeElement').pop();
-
-      // remove bands from canvas
-      canvas.removeShape(context.createdShapeA);
-      canvas.removeShape(context.createdShapeB);
-
-      // properly unlink elements regarding choreography tasks
-      if (is(shape, 'bpmn:ChoreographyTask')) {
-        unlinkMessageFlowSemantics(injector, shape.businessObject, context.createdMessageFlow);
-        canvas.removeShape(context.createdMessageShape);
+      if (!shape.oldBusinessObject) {
+        // remove the bands from the BPMNDiagram semantic object
+        let choreo = canvas.getRootElement().businessObject;
+        let bpmnDiagram = choreo.di;
+        bpmnDiagram.get('planeElement').pop();
+        bpmnDiagram.get('planeElement').pop();
+
+        // remove bands from canvas
+        canvas.removeShape(context.createdShapeA);
+        canvas.removeShape(context.createdShapeB);
+
+        // properly unlink elements regarding choreography tasks
+        if (is(shape, 'bpmn:ChoreographyTask')) {
+          unlinkMessageFlowSemantics(injector, shape.businessObject, context.createdMessageFlow);
+          canvas.removeShape(context.createdMessageShape);
+        }
+      } else {
+        // Delete messages, message flows, and band DIs that were created during copy paste
```

Comment:
Should this not be part of the revert logic in the copy paste handler? Seems inconsistent to put it here since the objects were not created here.

Reply:
It might also be sensible to use this as the default implementation for the revert of Choreos, as it does the same work as the code above, but in a more general way. Also in the copy paste revert it would then only be in case the element to revert is a choreo. However, I share the opinion that the current solution is not optimal.

Reply:
See issue  #100